app/src/main/java/com/android/sample/ui/createRecipe/AddInstructionStepScreen.kt

```diff
@@ -356,3 +355,27 @@ fun confirmAndAssignStep(
     onSuccess()
   }
 }
+
+fun navigateAfterDelete(
+    isEditing: Boolean,
+    navigationActions: NavigationActions,
+) {
+  val targetScreen =
+      if (isEditing) Screen.EDIT_RECIPE_LIST_INSTRUCTIONS
+      else Screen.CREATE_RECIPE_LIST_INSTRUCTIONS
+  val popUpToScreen =
+      if (isEditing) Screen.EDIT_CATEGORY_SCREEN else Screen.CREATE_RECIPE_LIST_INGREDIENTS
+  navigationActions.navigateToPop(targetScreen, popUpTo = popUpToScreen, inclusive = false)
+}
+
+fun navigateAfterValidation(
+    isEditing: Boolean,
+    navigationActions: NavigationActions,
+) {
+  val targetScreen =
+      if (isEditing) Screen.EDIT_RECIPE_LIST_INSTRUCTIONS
+      else Screen.CREATE_RECIPE_LIST_INSTRUCTIONS
+  val popUpToScreen =
+      if (isEditing) Screen.EDIT_CATEGORY_SCREEN else Screen.CREATE_RECIPE_LIST_INGREDIENTS
+  navigationActions.navigateToPop(targetScreen, popUpTo = popUpToScreen, inclusive = false)
+}
```

Comment:
Thank you. I put them as internal otherwise i couldnt test them directly even with the `@VisibleForTesting` tag as they arent in the same package. I ahve also added the documentation.